lib/knot.c

```diff
@@ -80,31 +80,32 @@ boolean k_read(Link **link, char *filename)
              over,
              i,
              j;
-  int        num_crossings;
+  int        num_crossings, pos;
   crossing  *kk;
   crossing   k[MAXCROSS];
-  FILE      *f;
 
   for (i = 0; i < MAXCROSS; i++) k[i].hand = 0;
 
-  f = fmemopen(filename, strlen(filename), "r");
   if (f == 0)
   {
-    fclose(f);
     return FALSE;
   }
-  fscanf(f, "%d ", &links);
+  sscanf(f, "%d%n", &links, &pos);
+  f += pos;
   for (i=0; i<links; ++i)                    /* how many pieces of string */
```

Comment:
This kind of updating the position of the f pointer could have side effects. Wouldn't it be safer to use a copy instead of the pointer passed to the function?